src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -246,34 +270,42 @@ public void run() {
     }
 
     public void enableServerDebug(int libertyDebugPort) throws IOException {
-        // creating jvm.options file to open a debug port
-        debug("jvmOptions: " + jvmOptions);
-        if (jvmOptions != null && !jvmOptions.isEmpty()) {
-            warn("Cannot start liberty:dev in debug mode because jvmOptions are specified in the server configuration");
-        } else {
-            File jvmOptionsFile = new File(serverDirectory.getAbsolutePath() + "/jvm.options");
-            if (jvmOptionsFile.exists()) {
-                debug("jvm.options already exists");
-                File jvmOptionsBackup = new File(serverDirectory.getAbsolutePath() + "/jvmBackup.options");
-                Files.copy(jvmOptionsFile.toPath(), jvmOptionsBackup.toPath());
-                boolean deleted = jvmOptionsFile.delete();
-                if (!deleted) {
-                    error("Could not move existing liberty:dev jvm.options file");
-                }
-            }
-            debug("Creating jvm.options file: " + jvmOptionsFile.getAbsolutePath());
-            StringBuilder sb = new StringBuilder("# Generated by liberty:dev \n");
-            sb.append("-Dwas.debug.mode=true\n");
-            sb.append("-Dcom.ibm.websphere.ras.inject.at.transform=true\n");
-            sb.append("-Dsun.reflect.noInflation=true\n");
-            sb.append("-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=" + libertyDebugPort);
-            BufferedWriter writer = new BufferedWriter(new FileWriter(jvmOptionsFile));
-            writer.write(sb.toString());
-            writer.close();
-            if (jvmOptionsFile.exists()) {
-                info("Successfully created liberty:dev jvm.options file");
-            }
-        }
+    	String serverEnvPath = serverDirectory.getAbsolutePath() + "/server.env";
+    	File serverEnvFile = new File(serverEnvPath);
+    	
+    	StringBuilder sb = new StringBuilder();
+    	if (serverEnvFile.exists()) {
+    		debug("server.env already exists");
+    		File serverEnvBackup = new File(serverEnvPath + ".bak");
+    		Files.copy(serverEnvFile.toPath(), serverEnvBackup.toPath());
+    		boolean deleted = serverEnvFile.delete();
+    		
+    		if (!deleted) {
+    			error("Could not move existing liberty:dev server.env file");
+    		}
+    		
+    		BufferedReader reader = new BufferedReader(new FileReader(serverEnvBackup));
+    		String line;
+    		while ((line = reader.readLine()) != null) {
+    			sb.append(line);
+    			sb.append("\n");
+    		}
+    		reader.close();
+    	}
+    	
+    	debug("Creating server.env file: " + serverEnvFile.getAbsolutePath());
+    	sb.append("WLP_DEBUG_SUSPEND=n\n");
+    	sb.append("WLP_DEBUG_ADDRESS=");
```

Comment:
what happens if the server.env already has WLP_DEBUG_ADDRESS defined?  Does the new one take precedence?

Reply:
Yes, the new one takes precedence @ericglau